libs/platform/core/src/orchestration/components/components-loader.ts

```diff
@@ -152,33 +155,32 @@ export class ComponentsLoader {
   ): Promise<ComponentType | void> {
     if (this.componentMap.has(def.name)) {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      return this.componentMap.get(def.name)!.observableType;
+      return this.componentMap.get(def.name)!.extendedClass;
     }
 
-    const [componentType, themes] = await Promise.all([
+    const [componentClass, themes] = await Promise.all([
       this.loadComponentImpl(def, meta),
       this.theme?.resolve(def),
     ]);
 
-    if (!componentType) {
+    if (!componentClass) {
       return;
     }
 
-    const observableType = observableShadow(componentType, def.name);
+    const extendedClass = componentExtender(componentClass, def.name);
 
     this.componentMap.set(def.name, {
-      observableType,
+      extendedClass,
       themes,
-      componentType,
     });
 
-    return observableType;
+    return extendedClass;
   }
 
   getComponentClass(
     tag: string
   ): (ComponentType & ComponentStatic) | undefined {
-    return this.componentMap.get(tag)?.componentType;
+    return Object.getPrototypeOf(this.componentMap.get(tag)?.extendedClass);
```

Comment:
`Object.getPrototypeOf` accepts objects only as parameter, but your expression contains optional chaining, that can cause violation.

Reply:
```
if (!this.componentMap.get(tag)?.extendedClass) return;
return Object.getPrototypeOf(this.componentMap.get(tag).extendedClass);
```